tests/integration/deregister-trailer.int.test.ts

```diff
@@ -0,0 +1,122 @@
+import supertest from 'supertest';
+import { emptyDatabase, populateDatabase } from '../utils/dbUtil';
+import * as domain from '../../src/domain';
+import * as trailerRegistrations from '../../data/trailer-registration.json';
+
+const url = 'http://localhost:3020/local';
+const request = supertest(url);
+
+describe('deregister trailer', () => {
+  beforeAll(async () => {
+    await emptyDatabase();
+  });
+
+  beforeEach(async () => {
+    await populateDatabase();
+  });
+
+  afterEach(async () => {
+    await emptyDatabase();
+  });
+
+  afterAll(async () => {
+    await populateDatabase();
+  });
+
+  test('should unregister when TRN is found', async () => {
```

Comment:
should we add a bit more description and tell what the service does? Maybe something along those lines
it is unregistering the 'record' and returns a response containing the relevant `vinOrChassisWithMake`, `reasonForDeregistration` fields/payload

Reply:
updated